src/Nethermind/Nethermind.Merge.Plugin.Test/PostMergeHeaderValidatorTests.cs

```diff
@@ -0,0 +1,322 @@
+        [TestCase(40000000, 5, 40039062, false)]
+        [TestCase(40000000, 5, 39960939, true)]
+        [TestCase(40000000, 5, 39960938, false)]
+        public void When_gaslimit_is_on_london_fork(long parentGasLimit, long blockNumber, long gasLimit, bool expectedResult)
+        {
+            OverridableReleaseSpec spec = new(London.Instance)
+            {
+                Eip1559TransitionBlock = 5
+            };
+            TestSpecProvider specProvider = new(spec);
+            _validator = new HeaderValidator(_blockTree, _ethash, specProvider, new OneLoggerLogManager(_testLogger));
+            _parentBlock = Build.A.Block.WithDifficulty(0)
+                            .WithGasLimit(parentGasLimit)
+                            .WithNumber(blockNumber)
+                            .TestObject;
+            _block = Build.A.Block.WithParent(_parentBlock)
+                .WithDifficulty(0)
+                .WithMixHash(new Keccak("0xd7db5fdd332d3a65d6ac9c4c530929369905734d3ef7a91e373e81d0f010b8e8"))
+                .WithGasLimit(gasLimit)
+                .WithNumber(_parentBlock.Number + 1)
+                .WithBaseFeePerGas(BaseFeeCalculator.Calculate(_parentBlock.Header, specProvider.GetSpec(_parentBlock.Number + 1)))
+                .WithNonce(0).TestObject;
+            _block.Header.SealEngineType = SealEngineType.None;
+            _block.Header.Hash = _block.CalculateHash();
+            
+            bool result = _validator.Validate(_block.Header, _parentBlock.Header);
+            Assert.AreEqual(expectedResult, result);
+        }
+        //Not good, I need to change the _validator to PostMergeHeaderValidator
+        //already fixed difficulty
+        [Test]
+        public void When_gas_limit_is_long_max_value()
+        {
+            _validator = new HeaderValidator(_blockTree, _ethash, _specProvider, new OneLoggerLogManager(_testLogger));
+            _parentBlock = Build.A.Block.WithDifficulty(0)
+                .WithGasLimit(long.MaxValue)
+                .WithNumber(5)
+                .TestObject;
+            _block = Build.A.Block.WithParent(_parentBlock)
+                .WithDifficulty(0)
+                .WithMixHash(new Keccak("0xd7db5fdd332d3a65d6ac9c4c530929369905734d3ef7a91e373e81d0f010b8e8"))
+                .WithGasLimit(long.MaxValue)
+                .WithNumber(_parentBlock.Number + 1)
+                .WithNonce(0).TestObject;
+            _block.Header.SealEngineType = SealEngineType.None;
+            _block.Header.Hash = _block.CalculateHash();
+            
+            bool result = _validator.Validate(_block.Header, _parentBlock.Header);
+            
+            Assert.True(result);
+        }
+        //I think this is good
+        [Test]
+        public void When_incorrect_nonce_then_invalid()
+        {
+            _block.Header.Nonce = 1;
+            _block.Header.SealEngineType = SealEngineType.None;
+            _block.Header.Hash = _block.CalculateHash();
+            
```

Comment:
please try to share the code for HeaderValidatorTests and PostMergeHeaderValidatorTests